plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/PartitionProjectionService.java

```diff
@@ -90,13 +91,12 @@ public PartitionProjectionService(
         this.projectionFactories = ImmutableMap.copyOf(requireNonNull(projectionFactories, "projectionFactories is null"));
     }
 
-    public Map<String, Object> getPartitionProjectionTrinoTableProperties(Table table)
+    public Map<String, Object> getPartitionProjectionTrinoTableProperties(HiveMetadata.TableParameterProvider tableParameterProvider)
```

Comment:
This can be a `static` method.

I'd argue that we should bring this method in `HiveMetadata` to have all `TableParameterProvider` handling in one place.